registry/token.go

```diff
@@ -51,10 +51,12 @@ func getToken(username, password string, params map[string]string, registryEndpo
 		reqParams.Add("scope", scopeField)
 	}
 
-	reqParams.Add("account", username)
+	if username != "" {
+		reqParams.Add("account", username)
+		req.SetBasicAuth(username, password)
+	}
 
 	req.URL.RawQuery = reqParams.Encode()
-	req.SetBasicAuth(username, password)
```

Comment:
@jlhawn putting this line inside the `if` is what's causing problems in integration tests. I haven't figure it all out yet but if you can think of a better way to fix the original issue while keeping this here, that'd be great too. Will be back later today to continue investigating.